src/FitterBase.cc

```diff
@@ -0,0 +1,221 @@
+        {
+            if (ms)  // measurement term
+                msOutliers.push_back(ms);
+            else  // ref term
+                fsOutliers.push_back(fs);
+            // mark the parameters as directly changed when we discard this chi2 term.
+            for (auto const &i : indices) affectedParams(i)++;
+            nOutliers++;
+        }
+    }  // end loop on measurements/references
+    LOGLS_INFO(_log, "findOutliers: found " << msOutliers.size() << " meas outliers and " << fsOutliers.size()
+                                            << " ref outliers ");
+
+    return nOutliers;
+}
+
+int FitterBase::minimize(const std::string &whatToFit, double nSigmaCut) {
+    assignIndices(whatToFit);
+
+    // return code can take 3 values :
+    // 0 : fit has converged - no more outliers
+    // 1 : still some ouliers but chi2 increases
+    // 2 : factorization failed
+    int returnCode = 0;
+
+    // TODO : write a guesser for the number of triplets
+    unsigned nTrip = (_lastNTrip) ? _lastNTrip : 1e6;
+    TripletList tripletList(nTrip);
+    Eigen::VectorXd grad(_nParTot);
+    grad.setZero();
+
+    // Fill the triplets
+    leastSquareDerivatives(tripletList, grad);
+    _lastNTrip = tripletList.size();
+
+    LOGLS_DEBUG(_log, "End of triplet filling, ntrip = " << tripletList.size());
+
+    SpMat hessian;
+    {
+        SpMat jacobian(_nParTot, tripletList.getNextFreeIndex());
+        jacobian.setFromTriplets(tripletList.begin(), tripletList.end());
+        // release memory shrink_to_fit is C++11
+        tripletList.clear();  // tripletList.shrink_to_fit();
+        hessian = jacobian * jacobian.transpose();
+    }  // release the Jacobian
+
+    LOGLS_DEBUG(_log, "Starting factorization, hessian: dim="
+                              << hessian.rows() << " non-zeros=" << hessian.nonZeros()
+                              << " filling-frac = " << hessian.nonZeros() / sqr(hessian.rows()));
+
+    CholmodSimplicialLDLT2<SpMat> chol(hessian);
+    if (chol.info() != Eigen::Success) {
+        LOGLS_ERROR(_log, "minimize: factorization failed ");
+        return 2;
+    }
+
+    unsigned totalOutliers = 0;
+    double oldChi2 = computeChi2().chi2;
+
+    while (true) {
+        Eigen::VectorXd delta = chol.solve(grad);
+        offsetParams(delta);
+        Chi2Statistic currentChi2(computeChi2());
+        LOGLS_DEBUG(_log, currentChi2);
+        if (currentChi2.chi2 > oldChi2) {
+            LOGL_WARN(_log, "chi2 went up, skipping outlier rejection loop");
+            returnCode = 1;
+            break;
+        }
+        oldChi2 = currentChi2.chi2;
+
+        if (nSigmaCut == 0) break;  // no rejection step to perform
+        MeasuredStarList msOutliers;
+        FittedStarList fsOutliers;
+        int nOutliers = findOutliers(nSigmaCut, msOutliers, fsOutliers);
+        totalOutliers += nOutliers;
+        if (nOutliers == 0) break;
+        TripletList tripletList(1000);  // initial allocation size.
```

Comment:
I'm pretty sure it is exactly the number of outliers computed above. Fixed, thanks.